server-common/src/main/java/org/apache/kafka/server/purgatory/DelayedOperation.java

```diff
@@ -41,19 +40,10 @@ public abstract class DelayedOperation extends TimerTask {
 
     private volatile boolean completed = false;
 
-    protected final Lock lock;
-
-    public DelayedOperation(long delayMs, Optional<Lock> lockOpt) {
-        this(delayMs, lockOpt.orElse(new ReentrantLock()));
-    }
+    protected final Lock lock = new ReentrantLock();
```

Comment:
Could you please change the type from `Lock` to `ReentrantLock`? This class requires some features of `ReentrantLock`.